working/1426-extension-types/feature-specification-views.md

```diff
@@ -58,12 +110,83 @@ requires a certain discipline in the use of `o`'s instance methods: We may
 call certain methods, but only in specific ways, and other methods should
 not be called at all. This kind of added discipline can be enforced by
 accessing `o` typed as a view type, rather than typed as its run-time
-type `R` or some supertype of `R` (which is what we normally do).
+type `R` or some supertype of `R` (which is what we normally do). For
+example:
+
+```dart
+view IdNumber(int i) {
+  // Declare a few members.
+
+  // Assume that it makes sense to compare ID numbers
+  // because they are allocated with increasing values,
+  // so "smaller" means "older".
+  operator <(IdNumber other) => i < (other as int);
```

Comment:
But not `other.i`?

Reply:
I hadn't thought about that, but I do think it would be useful to support that kind of access.

The developer who writes a view can choose a private name for the "wrapped object" (that is, the 'representation name') and disable this mechanism, but otherwise the representation name could be used as a getter that provides access to the representation object under its "original" type.

This is _better_ than casting, because there's no guarantee about a cast being correct (that is, we have no way to make `myIdNumber as int` pass but warn/err in case of `myIdNumber as String`).

The semantics is kind of funny, but it is all statically resolved, so I wouldn't expect the handling of `other.id` to introduce any special difficulties during code generation.

I adjusted the example to mention this.